src/drivers/src/SerialDetection.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * Created by: Emmanuel Sales
+ * Created on: February 18, 2017
+ * Description: This class is responsible for discovering application-specific arduinos 
+ *              by a particular fixed-length prefix at the beginning of each line 
+ *              the arduino broadcasts.
+ * 
+ */
+ 
+#include <SerialDetection.h>
+
+// Constructor for the class
+SerialDetector::SerialDetector() {
+    DIR *dir;
+    struct dirent *ent;
+    if ((dir = opendir("/dev/serial/by-id/")) != NULL) {
+        while((ent = readdir(dir)) != NULL) {
+            possibleSerialFilenames.push_back(std::string(ent->d_name));
+        }
+    } else {
+        throw direx;
+    }
+}
+
+// return an fstream of the arduino broadcasting the appropriate data
```

Comment:
Just generally, this comment should probably go in the header documentation I mentioned above.